# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1361864053 18000
#      Tue Feb 26 02:34:13 2013 -0500
# Node ID ea5d92f574e02da946e13b48a911ef64b547dd39
# Parent  3ec4f6488569ab6ceb55163d079010d49fe58b46
return yyparse status from octave_parse_input

* oct-parse.yy (octave_parse_input): Return status of parse.

diff --git a/libinterp/parse-tree/oct-parse.yy b/libinterp/parse-tree/oct-parse.yy
--- a/libinterp/parse-tree/oct-parse.yy
+++ b/libinterp/parse-tree/oct-parse.yy
@@ -4463,17 +4463,17 @@ cleanup_statement_list (tree_statement_l
       delete *lst;
       *lst = 0;
     }
 }
 
 int
 octave_parse_input (void)
 {
-  octave_parse ();
+  return octave_parse ();
 }
 
 DEFUN (eval, args, nargout,
   "-*- texinfo -*-\n\
 @deftypefn  {Built-in Function} {} eval (@var{try})\n\
 @deftypefnx {Built-in Function} {} eval (@var{try}, @var{catch})\n\
 Parse the string @var{try} and evaluate it as if it were an Octave\n\
 program.  If that fails, evaluate the optional string @var{catch}.\n\
